indico/web/templates/forms/location_widget.html

```diff
@@ -0,0 +1,534 @@
+                           type="checkbox">
+                    <label for="location-inheritance-{{ field.id }}">{% trans %}Use default{% endtrans %}</label>
+                </div>
+            </div>
+        </div>
+    </div>
+    <div class="clearfix"></div>
+{% endblock %}
+
+{% block javascript %}
+    <script>
+        (function() {
+            'use strict';
+
+            var field = $('#{{ field.id }}');
+            var description = field.closest('.form-field').find('.form-field-description');
+            var venueInput = $('#location-venue-{{ field.id }}');
+            var roomInput = $('#location-room-{{ field.id }}');
+            var address = $('#location-address-{{ field.id }}');
+            var usingDefault = $('#location-inheritance-{{ field.id }}');
+            var venues = {{ venues | tojson }};
+            var venueNames = $.map(venues.data, function(venue) {return venue.name;});
+            var rooms = {{ rooms | tojson }};
+            var hiddenData = field.val() ? JSON.parse(field.val()) : {};
+            var latestUsedField = '';
+            var latestSelectedOption = {field: '', value: ''};
+            var latestGroup = '';
+            var currentScrollPos = 0;
+            var delta = 0;
+            var validRoom = false;
+            var validVenue = false;
+            var preventGuessing = false;
+            var roomJustCleared = false;
+            var venueJustChanged = false;
+            var justUseOptionSelected = false;
+            var resultsWereOpen = false;
+
+            function initializeInputs() {
+                address.val(hiddenData.address);
+                roomInput.val(hiddenData.room_name);
+                venueInput.val(hiddenData.venue_name);
+                if (hiddenData.venue_id) {
+                    highlightOption(venueInput, venueInput.val());
+                }
+                if (hiddenData.room_id) {
+                    highlightOption(roomInput, roomInput.val());
+                }
+                usingDefault.prop('checked', hiddenData.inheriting);
+            }
+
+            function toggleDefaultLocation(isChecked, isInitStep) {
+                var message = ($T.gettext('Using default location from: ') + {{ parent[1] | tojson }} +
+                               ' (' + {{ parent[0] | tojson }} + ')');
+                var text = isChecked ? message : $T.gettext('Using custom location');
+                description.html(text);
+                venueInput.prop('disabled', isChecked).toggleClass('disabled', isChecked);
+                roomInput.prop('disabled', isChecked).toggleClass('disabled', isChecked);
+                address.prop('disabled', isChecked).toggleClass('disabled', isChecked);
+                field.closest('.i-location-field').toggleClass('disabled', isChecked);
+
+                hiddenData.inheriting = isChecked;
+                field.val(JSON.stringify(hiddenData)).trigger('change');
+
+                if (!isInitStep) {
```

Comment:
It's a piece of code that should run in every click of the 'use-default' checkbox, however, it should not be executed on page load where I manually trigger the click. I thought this was the simplest way to solve it.